src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/BehaviourService.kt

```diff
@@ -14,48 +14,46 @@ class BehaviourService(
   private val prisonApiService: PrisonApiService,
   private val clock: Clock,
 ) {
-  private fun behaviourCaseNoteMap() = mapOf("POS" to "IEP_ENC", "NEG" to "IEP_WARN")
+  private val behaviourCaseNoteMap = mapOf("POS" to "IEP_ENC", "NEG" to "IEP_WARN")
 
-  suspend fun getBehaviours(reviews: List<PrisonerIepLevel>) =
-    getCaseNoteUsageByLastReviewDate(
+  suspend fun getBehaviours(reviews: List<PrisonerIepLevel>): Map<BookingTypeKey, CaseNoteSummary> {
+    val lastRealReviews = getLastRealReviewForOffenders(reviews)
+    val lastReviewsOrDefaultPeriods = lastRealReviews.mapValues { truncateReviewDate(it.value) }
+    return getCaseNoteUsageByLastReviewDate(
       prisonApiService.retrieveCaseNoteCountsByFromDate(
-        behaviourCaseNoteMap().keys.toList(),
-        getLastRealReviewForOffenders(reviews)
+        behaviourCaseNoteMap.keys.toList(),
+        lastReviewsOrDefaultPeriods,
       ).toList(),
-      behaviourCaseNoteMap(),
     )
-  private fun getCaseNoteUsageByLastReviewDate(caseNotesByType: List<PrisonerCaseNoteByTypeSubType>, subTypeMap: Map<String, String>) =
+  }
+
+  private fun getCaseNoteUsageByLastReviewDate(caseNotesByType: List<PrisonerCaseNoteByTypeSubType>) =
     caseNotesByType
       .groupBy(PrisonerCaseNoteByTypeSubType::toKey)
-      .map { cn ->
+      .mapValues { cn ->
         CaseNoteSummary(
           key = cn.key,
           totalCaseNotes = calcTypeCount(cn.value),
-          numSubTypeCount = calcTypeCount(cn.value.filter { cnc -> cnc.caseNoteSubType == subTypeMap[cn.key.caseNoteType] })
-
+          numSubTypeCount = calcTypeCount(cn.value.filter { cnc -> cnc.caseNoteSubType == behaviourCaseNoteMap[cn.key.caseNoteType] }),
         )
-      }.associateBy { it.key }
+      }
 
   private fun calcTypeCount(caseNoteUsage: List<PrisonerCaseNoteByTypeSubType>): Int =
-    caseNoteUsage.map { it.numCaseNotes }.fold(0) { acc, next -> acc + next }
+    caseNoteUsage.sumOf { it.numCaseNotes }
 
-  private fun getLastRealReviewForOffenders(reviews: List<PrisonerIepLevel>): Map<Long, LocalDateTime> =
+  private fun getLastRealReviewForOffenders(reviews: List<PrisonerIepLevel>) =
```

Comment:
this now actually returns real review dates so they can be null: truncation to at-most-last-3-months is now done where the prison-api call is made